contracts/asset_manager/src/contract.rs

```diff
@@ -31,18 +29,27 @@ pub struct AssetManager;
 
 #[contractimpl]
 impl AssetManager {
-    pub fn initialize(env: Env, registry: Address, admin: Address, config: ConfigData)  -> Result<(), ContractError> {
-        if has_registry(&env.clone()) {
+    pub fn initialize(env: Env, admin: Address, xcall: Address, xcall_manager: Address, native_address: Address, icon_asset_manager: String, upgrade_authority: Address)  -> Result<(), ContractError> {
+        if has_upgrade_authority(&env.clone()) {
             return Err(ContractError::ContractAlreadyInitialized)
         }
-        write_registry(&env, &registry);
         write_administrator(&env, &admin);
-        Self::configure(env, config);
+        states::write_xcall(&env, xcall);
+        states::write_xcall_manager(&env, xcall_manager);
+        states::write_icon_asset_manager(&env, icon_asset_manager);
+        states::write_upgrade_authority(&env, upgrade_authority);
+        states::write_native_address(&env, native_address);
         Ok(())
     }
 
-    pub fn get_config(env: Env) -> ConfigData {
-        get_config(&env)
+    pub fn get_config(env: Env) -> (Address, Address, Address, String, Address) {
+        (
+            states::read_xcall(&env),
+            states::read_xcall_manager(&env),
+            states::read_native_address(&env),
+            states::read_icon_asset_manager(&env),
+            states::read_upgrade_authority(&env),
+        )
     }
```

Comment:
since we are splitting storage items , its better to have separate getter methods as well. reading all storage item just to get one is unnecessary . separate getters would be more readable as well. If we may need entire config object , return type still can be 'config' struct. makes it more readable and type safe.

Reply:
yeah returned using struct